web/src/main/java/com/microsoft/azure/simpletodo/model/TodoItem.java

```diff
@@ -4,28 +4,34 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import java.time.OffsetDateTime;
 
-import org.springframework.data.annotation.Id;
+import org.hibernate.annotations.GenericGenerator;
 import org.springframework.format.annotation.DateTimeFormat;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
 import javax.validation.Valid;
 import javax.validation.constraints.*;
 
 
-import java.util.*;
 import javax.annotation.Generated;
 
 /**
  * A task that needs to be completed
  */
 
 @Generated(value = "org.openapitools.codegen.languages.SpringCodegen")
+@Entity
 public class TodoItem {
 
   @JsonProperty("id")
   @Id
-  private UUID id;
+  @GeneratedValue(generator="system-uuid")
+  @GenericGenerator(name="system-uuid", strategy = "uuid")
+  private String id;
```

Comment:
this is too complicated, let's use `Long` and `GeneratedValue`, no need to config the id generator or hibernate API, only jpa api

Reply:
Actually, I tried. After changed to `Long`, it keeps sending http request from front end to back end. I tried to fix the problem by updating the code in `client` module but failed. Here is the screenshot about the id in `client` module:
> ![image](https://user-images.githubusercontent.com/13167207/219587598-a7798bf0-f6e8-40e2-b172-8cc070f87532.png)

OK. I'll try again. Change the primary key type to `Long` in both `client` and `web`.

Reply:
this is exactly the scenario we need to solve: does frontend determine what the entity model is like?
how we resolve this in the real-world ?

Reply:
> does frontend determine what the entity model is like?

No, frontend and backend should get an agreement before start development.

> how we resolve this in the real-world ?

In my understanding, we should follow these steps:
1. Define the REST API.
2. Generate the template code (by tool like swagger) in both frontend and back end.
3. Implement business logic in both frontend and backend.

I added a new commit: https://github.com/Azure-Samples/ASA-Samples-Web-Application/pull/6/commits/2bef789c2bf3ac48a7babcd618b5725d2edefdb3 which keep the `id` field's type as `string` by adding this annotation:
> `@JsonSerialize(using = ToStringSerializer.class)`.